openmc/mgxs/mdgxs.py

```diff
@@ -388,19 +410,53 @@ def get_xs(self, groups='all', subdomains='all', nuclides='all',
 
         # Reshape tally data array with separate axes for domain, energy groups,
         # delayed groups, and nuclides
-        num_subdomains = int(xs.shape[0] / (num_groups * num_delayed_groups))
-        new_shape = (num_subdomains, num_delayed_groups, num_groups)
-        new_shape += xs.shape[1:]
-        xs = np.reshape(xs, new_shape)
+        # Accomodate the polar and azimuthal bins if needed
+        if self.num_polar or self.num_azimuthal:
+            if self.num_polar:
+                num_pol = self.num_polar
+            else:
+                num_pol = 1
+            if self.num_azimuthal:
+                num_azi = self.num_azimuthal
+            else:
+                num_azi = 1
+            num_subdomains = int(xs.shape[0] /
+                                 (num_groups * num_delayed_groups *
+                                  num_pol * num_azi))
+            new_shape = (num_pol, num_azi, num_subdomains, num_delayed_groups,
+                         num_groups) + xs.shape[1:]
+            xs = np.reshape(xs, new_shape)
+
+            # Reverse data if user requested increasing energy groups since
+            # tally data is stored in order of increasing energies
+            if order_groups == 'increasing':
+                xs = xs[:, :, :, :, ::-1, :]
+
+            if squeeze:
+                # We want to squeeze out everything but the polar, azimuthal,
+                # delayed group, and energy group data.
+                dont_squeeze = (0, 1, 3, 4)
+                # Squeeze will return a ValueError if the axis has a size
+                # greater than 1, so try each axis in axes one at a time,
+                # and do our own check to preclude the ValueError
+                initial_shape = len(xs.shape)
+                for axis in range(initial_shape - 1, -1, -1):
+                    if axis not in dont_squeeze and xs.shape[axis] == 1:
+                        xs = np.squeeze(xs, axis=axis)
+        else:
+            num_subdomains = int(xs.shape[0] / (num_groups * num_delayed_groups))
```

Comment:
Add back in the comment for this block about which dimensions are included (and expand the comment to include angle).